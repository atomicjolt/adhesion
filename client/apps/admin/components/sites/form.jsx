import React from 'react';
import PropTypes from 'prop-types';
import Input from '../common/input';

export const FIELDS = {
  oauth_key: 'Canvas Developer ID',
  oauth_secret: 'Canvas Developer Key',
  url: 'Canvas Domain',
};

const SiteForm = (props) => {
  const {
    isUpdate,
    oauth_key,
    oauth_secret,
    url,
    onChange,
    setupSite,
    closeModal,
  } = props;
  const buttonVerb = isUpdate ? 'Update' : 'Create';

  const inputTemplate = (fieldLabel, field, value) => {
    const inputProps = {
      id: `site_${field}`,
      name: field,
      type: 'text',
      value: value || '',
      onChange,
    };
    return (
      <div key={field} className="o-grid__item u-full">
        <Input
          className="c-input"
          labelText={fieldLabel}
          inputProps={inputProps}
        />
      </div>
    );
  };

  return (
    <form>
      <div className="o-grid o-grid__modal-top">
        {inputTemplate(FIELDS.url, 'url', url)}
        {inputTemplate(FIELDS.oauth_key, 'oauth_key', oauth_key)}
        {inputTemplate(FIELDS.oauth_secret, 'oauth_secret', oauth_secret)}
      </div>


      <button
        type="button"
        className="c-btn c-btn--yellow"
        onClick={setupSite}
      >
        {buttonVerb} Domain
      </button>
      <button
        type="button"
        className="c-btn c-btn--gray--large u-m-right"
        onClick={closeModal}
      >
        Cancel
      </button>
    </form>
  );
};

SiteForm.propTypes = {
  setupSite: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  isUpdate: PropTypes.bool.isRequired,
  oauth_key: PropTypes.string,
  oauth_secret: PropTypes.string,
  url: PropTypes.string,
};

export default SiteForm;
