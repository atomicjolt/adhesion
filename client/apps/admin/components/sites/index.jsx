import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Heading from '../common/heading';
import Header from './header';
import SiteModal from './modal';
import List from './list';
import { deleteSite } from '../../actions/sites';

function select(state) {
  return {
    sites: state.sites,
  };
}

export class Index extends React.Component {
  static propTypes = {
    sites: PropTypes.shape({}).isRequired,
    deleteSite: PropTypes.func.isRequired,
  }

  constructor() {
    super();
    this.state = {
      siteModalOpen: false,
    };
  }

  closeSiteModal() {
    this.setState({
      siteModalOpen: false,
    });
  }

  render() {
    return (
      <div>
        <Heading />
        <div className="o-contain o-contain--full">
          <SiteModal
            isOpen={this.state.siteModalOpen}
            closeModal={() => this.closeSiteModal()}
          />
          <Header
            openSettings={() => {}}
            newSite={() => this.setState({ siteModalOpen: true })}
          />
          <List
            sites={this.props.sites}
            deleteSite={this.props.deleteSite}
          />
        </div>
      </div>
    );
  }
}

export default connect(select, { deleteSite })(Index);
