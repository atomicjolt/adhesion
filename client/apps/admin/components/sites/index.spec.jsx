import React from 'react';
import { shallow } from 'enzyme';
import { Index } from './index';
import sites from '../../reducers/sites';

jest.mock('../../libs/assets');
describe('sites index', () => {
  let result;
  let props;

  const sitesData = {
    1: {
      id: 1,
      url: 'bfcoder.com'
    },

    2: {
      id: 2,
      url: 'atomicjolt.com'
    }
  };

  beforeEach(() => {
    props = {
      sites: sitesData,
    };
    result = shallow(<Index {...props} />);
  });

  it('renders', () => {
    expect(result).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
