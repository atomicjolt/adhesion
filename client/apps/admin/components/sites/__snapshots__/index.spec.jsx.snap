// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sites index matches the snapshot 1`] = `
<div>
  <Connect(Heading) />
  <div
    className="o-contain o-contain--full"
  >
    <Connect(SiteModal)
      closeModal={[Function]}
      isOpen={false}
    />
    <Header
      newSite={[Function]}
      openSettings={[Function]}
    />
    <List
      deleteSite={[Function]}
      sites={
        Object {
          "1": Object {
            "id": 1,
            "url": "bfcoder.com",
          },
          "2": Object {
            "id": 2,
            "url": "atomicjolt.com",
          },
        }
      }
    />
  </div>
</div>
`;
